qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -346,6 +347,8 @@ def _connect_tab_signals(self, tab):
             functools.partial(self._on_title_changed, tab))
         tab.icon_changed.connect(
             functools.partial(self._on_icon_changed, tab))
+        tab.search_match_changed.connect(
+            functools.partial(self._on_search_match_changed, tab))
```

Comment:
This should go through `self._filter` like above, so that it ensures that only search match changes from the current tab are processed. Also, the signal should be named `cur_search_match_changed` then.